charts/telemetry-collector/templates/configmap.yaml

```
@@ -0,0 +1,78 @@
+{{- include
+      "accelleran.common.configMap"
+      (fromYaml (include "accelleran.telemetry-collector.configMap.args" .))
+-}}
+
+{{- define "accelleran.telemetry-collector.configMap.args" -}}
+{{- $ := . -}}
+
+top:
+  {{ $ | toYaml | nindent 2 }}
+
+name: {{ include "accelleran.common.fullname" (dict "top" $) }}-config
+
+data:
+  config.yaml: |-
+    internal:
+      kafka:
+        url: "{{ include "accelleran.common.bootstrap.kafka.hostname" (dict "top" .) }}:{{ include "accelleran.common.bootstrap.kafka.port" (dict "top" .) }}"
+      {{- if $.Values.config.influxDB }}
+      influxDB:
+        {{- if $.Values.config.influxDB.url }}
+        url: "{{ $.Values.config.influxDB.url }}"
+        {{ else }}
+        url: "http://{{ $.Release.Name }}-influxdb2 }}"
+        {{- end }}
+        org: "{{ $.Values.config.influxDB.org }}"
+      {{- end }}
+    pipelines:
+      {{- $natsHostname := include "accelleran.common.bootstrap.nats.hostname" (dict "top" .) }}
+      {{- $natsPort := include "accelleran.common.bootstrap.nats.port" (dict "top" .) }}
+      {{- range $pipeline := $.Values.config.pipelines }}
+      - name: "{{ $pipeline.name }}" 
+        component: "{{ $pipeline.component }}"
+        vendor: "{{ $pipeline.vendor }}"
+        enabled: "{{ $pipeline.enabled }}"
+        {{- if $pipeline.consumers }}
+        consumers:
+          {{- if $pipeline.consumers.natsConsumer }}
+            natsConsumer:
+              {{- if $pipeline.consumers.natsConsumer.url }}
+              url: "{{ $pipeline.consumers.natsConsumer.url }}"
+              {{ else }}
+              url: "{{ $natsHostname }}:{{ $natsPort }}"
+              {{- end }}
```

I would use the same pattern as with influxdb2:

```suggestion
              url: {{ $pipeline.consumers.natsConsumer.url | default (printf "%s:%v" $natsHostname $natsPort) | quote }}
```

(I'm using `%v` iso `%d` or `%s` as normally it is a string here with the template call from before. But maybe this at some point could change? :shrug:)

---

OK